spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindHiddenMethod.java

```diff
@@ -0,0 +1,143 @@
+     * Thereof, it is useless to analyze super interfaces in the detector.
+     */
+    @Override
+    public void visitClassContext(ClassContext classContext) {
+        //First, I get the subclass and superclasses in variables.
+        //This is the current class. Named it subClass to better depict the idea of sub and super class.
+        JavaClass subClass = classContext.getJavaClass();
+        //No need for null check as every class has at least `Object` as its super class.
+        JavaClass[] superClasses;
+        try {
+            superClasses = subClass.getSuperClasses();
+        } catch (ClassNotFoundException e) {
+            AnalysisContext.reportMissingClass(e);
+            return;
+        }
+        //I store all the methods of subclass in an array
+        Method[] methods = subClass.getMethods();
+        //I filter the non-private and static methods, as private methods can't be overridden.
+        //Then for each method, I go through each method of the superclass for checking method hiding.
+        for (Method method : methods) {
+            //Careful!!! regarding the order of the conditions applied here
+            //Taking advantage of short circuit evaluation here by placing !method.isStatic() first.
+            //I check for the exceptional cases of inner class using three auxiliary private methods.
+            if (!method.isStatic() || method.isPrivate() || isMainMethod(method) ||
+                    isConstructor(method) || isHidingInnerClass(method) || isAutoGeneratedMethod(method)) {
+                continue;
+            }
+
+            for (JavaClass superClass : superClasses) {
+                Method[] superMethods = superClass.getMethods();
+                //I check whether the subclass method is hiding the superclass method. If yes I report the bug.
+                if (Arrays.asList(superMethods).contains(method)) {
+                    bugReporter.reportBug(new BugInstance(this, "HSM_HIDING_METHOD", NORMAL_PRIORITY)
+                            .addClass(subClass.getClassName())
+                            .addMethod(subClass, method)
+                            .addClass(superClass.getClassName())
+                            .addSourceLine(SourceLineAnnotation.fromVisitedInstruction(subClass, method, 0)));
+                }
+            }
+        }
+    }
+
+    @Override
+    public void report() {
+    }
+
+    /**
+     *This method checks for the inner class exceptional cases.
+     *As whenever there an inner class, '.access$' methods are created hiddenly to access the outer class attributes.
+     */
+    private boolean isHidingInnerClass(Method method) {
+        return method.getName().contains("access$");
+    }
+
+    /**
+     *This method checks for the autoGenerated methods for an inner class - exceptional cases.
+     *As whenever there an inner class, '.class$' methods are created hiddenly.
+     */
```

Comment:
Can you please add spaces in the javadoc after the `*`s like this?

```suggestion
    /**
     * This method checks for the autoGenerated methods for an inner class - exceptional cases.
     * As whenever there an inner class, '.class$' methods are created hiddenly.
     */
```